agent/rpc/peering/service.go

```diff
@@ -171,6 +171,11 @@ func (s *Server) GenerateToken(
 		return resp, err
 	}
 
+	// validate that this peer name is not being used as a PeerRole_DIALER
+	if err = s.validatePeerRole(req.PeerName, req.Partition, pbpeering.PeerRole_ACCEPTOR); err != nil {
+		return nil, err
+	}
+
```

Comment:
The `GetAgentCACertificates` and `GetServerAddresses` calls below can be moved a bit closer to where they're actually used. That way if these validations fail we don't make unnecessary backend calls.

Reply:
Moved those on top of this call